### PR TITLE
Create code to run in a pipeline operation

### PR DESCRIPTION
**Pull request by samaloney**
* Added code to watch a directory for new TM filess
* These tm files are then processed from TM to LB, L0 and L1
* Moved some config into a `stixconfig.ini` file and reader

### CODE REVIEW

File: stixcore/processing/pipeline.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            raise TypeError('func must be a callable')
+        self.func = func
+
+    def on_moved(self, event):
+        if TM_REGEX.match(event.dest_path):
+            self.func(Path(event.dest_path))
+
+
+def process(path):
+    lb_files = process_tmtc_to_levelbinary([SOCPacketFile(path)])
```
Comment (samaloney) [inline stixcore/processing/pipeline.py:50]: Not sure if the list of file returned will be unique - maybe need to create set before they are returned here

File: stixcore/io/fits/processors.py [hunk 2]
```diff
@@ -491,11 +490,21 @@ def write_fits(self, product):
...
+                energies['e_low'] = elow * u.keV
+                energies['e_high'] = ehigh * u.keV
+
+                energy_enc = fits.connect._encode_mixins(energies)
+                energy_hdu = table_to_hdu(energy_enc)
+                energy_hdu.name = 'ENERGIES'
+                hdus.append(energy_hdu)
 
-            hdul = fits.HDUList([primary_hdu, control_hdu, data_hdu, idb_hdu, energy_hdu])
+            hdul = fits.HDUList(hdus)
```
Comment (nicHoch) [inline stixcore/io/fits/processors.py:508]: was all this energy updates not already part of another PR

File: stixcore/processing/pipeline.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    l0_proc = Level0(CONFIG.get('Paths', 'tm_archive'), CONFIG.get('Paths', 'fits_archive'))
+    l0_files = l0_proc.process_fits_files(files=lb_files)
+    l1_proc = Level1(CONFIG.get('Paths', 'tm_archive'), CONFIG.get('Paths', 'fits_archive'))
+    l1_files = l1_proc.process_fits_files(files=l0_files)
+    logger.debug(l1_files)
```
Comment (nicHoch) [inline stixcore/processing/pipeline.py:55]: here somewhere we might have to "update" the latest SpiceKernel

**Review by nicHoch — APPROVED** [review]
i think the direction is clear to me.
looking forward so see it in action

File: stixcore/io/fits/processors.py (hunk 2 above)
Comment (samaloney) [reply stixcore/io/fits/processors.py:508]: Yea it was but it assumed the energy extension was always present I just made it optional.

File: stixcore/processing/pipeline.py (hunk 3 above)
Comment (samaloney) [reply stixcore/processing/pipeline.py:55]: Yea it depends on our strategy the configuration file can also be written so could update when I new spice kernel arrives or just always use the most recent meta kernel?

[the remark this answers is not in the document]
Comment (samaloney) [reply stixcore/processing/pipeline.py:55]: where should we move it to the data archive under a TM folder?

**Pull request merged**